RedacteurPortaal.Grains/Grains/ArchiveGrain.cs

```diff
@@ -17,37 +20,203 @@ public class ArchiveGrain : Grain, IArchiveGrain
+        if (audioItem == null)
+        {
+            throw new KeyNotFoundException();
+        }
+        else
+        {
+            return await Task.FromResult(audioItem);
+        }
     }
 
-    public async Task AddVideoItem(MediaVideoItem videoItem)
+    public async Task<NewsItemModel> GetNewsItem(Guid guid)
+    {
+        var newsItems = this.archive.State.NewsItems;
+        var newsItem = newsItems.Find(x => x.Id.Equals(guid));
+        if (newsItem == null)
+        {
+            throw new KeyNotFoundException();
+        }
+        else
+        {
+            return await Task.FromResult(newsItem);
+        }
+    }
+
+    public async Task<ArchiveModel> CreateArchive(ArchiveModel archive)
+    {
+        this.archive.State = archive;
+        await this.archive.WriteStateAsync();
+        return await Task.FromResult(this.archive.State);
+    }
+
+    public async Task<MediaVideoItem> AddVideoItem(MediaVideoItem videoItem)
     {
         this.archive.State.MediaVideoItems.Add(videoItem);
         await this.archive.WriteStateAsync();
+        return await Task.FromResult(videoItem);
     }
 
-    public async Task AddPhotoItem(MediaPhotoItem photoItem)
+    public async Task<MediaPhotoItem> AddPhotoItem(MediaPhotoItem photoItem)
     {
         this.archive.State.MediaPhotoItems.Add(photoItem);
         await this.archive.WriteStateAsync();
+        return await Task.FromResult(photoItem);
     }
 
-    public async Task AddAudioItem(MediaAudioItem audioItem)
+    public async Task<MediaAudioItem> AddAudioItem(MediaAudioItem audioItem)
     {
         this.archive.State.MediaAudioItems.Add(audioItem);
         await this.archive.WriteStateAsync();
+        return await Task.FromResult(audioItem);
     }
 
-    public async Task Update(ArchiveModel archive)
+    public async Task<NewsItemModel> AddNewsItem(NewsItemModel newsItem)
     {
-        this.archive.State = archive;
+        this.archive.State.NewsItems.Add(newsItem);
         await this.archive.WriteStateAsync();
+        return await Task.FromResult(newsItem);
     }
 
-    public async Task<ArchiveModel> Get()
+    public async Task Delete()
+    {
+        await this.archive.ClearStateAsync();
+    }
+
+    public async Task DeleteVideoItem(Guid videoItemId)
+    {
+        var videoItems = this.archive.State.MediaVideoItems;
+        if (videoItems != null && videoItems.Count > 0)
+        {
+            var videoItem = videoItems.Find(x => x.Id.Equals(videoItemId));
+            if (videoItem == null)
+            {
+                throw new KeyNotFoundException("Video item with id not found");
+            }
+            else
+            {
+                this.archive.State.MediaVideoItems.Remove(videoItem);
+                await this.archive.WriteStateAsync();
+            }
+        }
+        else
+        {
+            throw new KeyNotFoundException("List of video items with given archive id not found");
```

Comment:
Is er een scenario waar een archive geen videoitemlist heeft? hij kan leeg zijn maar bestaat toch altijd wel?

Reply:
ja idd, ik verander dit naar de tekst : "List of videoitems empty in given archideId"